src/auto-evo/Miche.cs

```diff
@@ -0,0 +1,210 @@
+﻿namespace AutoEvo;
+
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+///   A node for the Miche Tree
+/// </summary>
+/// <remarks>
+///   <para>
+///     The Miche class forms a tree by storing a list of child instances of Miche Nodes. If a Miche has no children
+///     it is considered a leaf node and can have a species Occupant instead. This class handles insertion into the
+///     tree through scores from the selection pressure it contains. For a fuller explanation see docs/auto_evo.md
+///   </para>
+/// </remarks>
+public class Miche
+{
+    public Miche? Parent;
+    public List<Miche> Children = new();
```

Comment:
I think it would be good to have this and `Pressure` be read only or is there a valid use case of these being changed after the miche is created?

Reply:
No, I agree that making this readonly would be good.

Reply:
After rereading, I will note that Parent and children can't be readonly due to the way the Miche Tree is constructed. It is a lot easier to do outside of a constructor function to avoid like 6 layers of nesting. I was agreeing that Pressure should be readonly.

Reply:
Note that making the list `readonly` only prevents reassigning the list, modifying the contents will still be possible even if the variable is readonly.